web/app/js/index.js

```diff
@@ -32,29 +31,62 @@ if (proxyPathMatch) {
   pathPrefix = proxyPathMatch[0];
 }
 
-const context = {
-  ...appData,
-  api: ApiHelpers(pathPrefix),
-  pathPrefix: pathPrefix,
-  productName: "Linkerd"
-};
+let selectedNamespace = "default";
+let pathArray = window.location.pathname.split("/");
+if (pathArray[0] === "" && pathArray[1] === "namespaces" && pathArray[2]) {
+  selectedNamespace = pathArray[2];
+}
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      ...appData
+    };
 
-const theme = createMuiTheme(dashboardTheme);
+    this.state.api = ApiHelpers(pathPrefix);
+    this.state.pathPrefix = pathPrefix;
+    this.state.productName = "Linkerd";
+    this.state.selectedNamespace = selectedNamespace;
 
-let applicationHtml = (
-  <React.Fragment>
-    <CssBaseline />
-    <MuiThemeProvider theme={theme}>
-      <AppContext.Provider value={context}>
+    this.state.updateNamespaceInContext = name => {
+      this.setState({selectedNamespace:name});
+    };
+  }
+
+  render() {
+    return (
+      <AppContext.Provider value={this.state}>
+        <AppHTML />
+      </AppContext.Provider>
+    );
+  }
+}
+
+
+function AppHTML() {
+  const theme = createMuiTheme(dashboardTheme);
+
+  return (
+    <React.Fragment>
+      <CssBaseline />
+      <MuiThemeProvider theme={theme}>
         <BrowserRouter>
           <RouterToUrlQuery>
             <Switch>
-              <Redirect exact from={`${pathPrefix}/`} to={`${pathPrefix}/overview`} />
-              <Route
-                path={`${pathPrefix}/overview`}
-                render={props => <Navigation {...props} ChildComponent={NamespaceLanding} />} />
+              <Redirect exact from={`${pathPrefix}/`} to={`${pathPrefix}/namespaces`} />
+              <Redirect exact from={`${pathPrefix}/overview`} to={`${pathPrefix}/namespaces`} />
+              <Redirect exact from={`${pathPrefix}/deployments`} to={`${pathPrefix}/namespaces`} />
+              <Redirect exact from={`${pathPrefix}/trafficsplits`} to={`${pathPrefix}/namespaces`} />
+              <Redirect exact from={`${pathPrefix}/daemonsets`} to={`${pathPrefix}/namespaces`} />
+              <Redirect exact from={`${pathPrefix}/statefulsets`} to={`${pathPrefix}/namespaces`} />
+              <Redirect exact from={`${pathPrefix}/jobs`} to={`${pathPrefix}/namespaces`} />
+              <Redirect exact from={`${pathPrefix}/replicationcontrollers`} to={`${pathPrefix}/namespaces`} />
+              <Redirect exact from={`${pathPrefix}/pods`} to={`${pathPrefix}/namespaces`} />
```

Comment:
Why not instead of redirecting all the legacy links to `/namespaces`, redirect to their new locations? E.g.
```
<Redirect exact from={`${pathPrefix}/deployments`} to={`${pathPrefix}/namespaces/all/deployments`} />
```

Reply:
Building on that idea, there's a weird edge case here where a cluster could have a namespace called `all`. To handle this, perhaps the legacy links should become the true `all` pages:
`/deployments` => deployments in all namespaces
`/namespaces/all/deployments` => deployments in the `all` namespace

Reply:
These are great comments, thanks @siggy @alpeb for thinking about this.

My thought on URLs: Since we are now querying the API by specific namespace (with a voluntary `all` option), we want to formulate our URLs to teach the users that. I like having an `all` in the URL to explicitly communicate what the user is looking at. 

I checked the K8s dashboard and URLs are not as user-friendly but they do always explicitly state the namepace - here's the `Jobs` workload page URL when I have `all namespaces` selected: `/proxy/#!/job?namespace=_all`

I agree that it might make sense to redirect `/deployments/` to `/namespaces/all/deployments` instead of redirecting to the home view (`/namespaces`) as I'd initially thought, I can see how that could be confusing.

Reply:
@siggy I just re-read and see what you mean re: a namespace explicitly named `all`. Since you could name a namespace absolutely anything, I see what you mean. I like the URL pattern of `/namespaces/all/deployments` but if it couldn't handle every possible namespace, we should probably go back to `/deployments`.

Reply:
One more idea, we could have the default path be `/namespaces/all_namespaces/deployments` since you can't have an `_` in a K8s namespace name.

Reply:
@scottcarol Yup! I see the k8s dashboard uses `_all`, that would work fine, or `all_namespaces`.

Reply:
My preference is `_all` like the k8s dashboard.

Reply:
Sounds good, this is my understanding of our conversation, thanks all:

* list of pods in all namespaces:       `/namespaces/_all/pods`
* list of pods in linkerd namespace: `/namespaces/linkerd/pods`
* `/pods` will redirect to                    `/namespaces/_all/pods`